src/app/services/ckan.service.ts

```diff
@@ -2,70 +2,96 @@ import { Injectable } from '@angular/core';
 import { HttpClient } from '@angular/common/http';
 import { Observable, catchError, map, of, from, mergeMap, toArray } from 'rxjs';
 import { environment } from 'src/environment/environment';
-import { Dataset } from '../interfaces/dataset-details.interface';
+import { Dataset } from '../interfaces/dataset-details';
+import { CatalogueDetail } from '../interfaces/catalogue-details';
+import { PartialDataset } from '../interfaces/dataset';
 
 @Injectable({
-  providedIn: 'root'
+  providedIn: 'root',
 })
 export class CkanService {
-  ckanToDcat: Map<string, string> = new Map(Object.entries({ package: "Dataset", organization: "Catalogue", tag: "Keyword", group: "Theme" }));
+  private static readonly ckanToDcat: Map<string, string> = new Map(
+    Object.entries({
+      package: 'Dataset',
+      organization: 'Catalogue',
+      tag: 'Keyword',
+      group: 'Theme',
+    })
+  );
 
-  constructor(private http: HttpClient) { }
+  constructor(private http: HttpClient) {}
 
-  searchDatasets(query: string, start: number = 0, rows: number = 12): Observable<{ results: any[], count: number }> {
-    const url = `${environment.backendUrl}/api/3/action/package_search?q=${encodeURIComponent(query)}&start=${start}&rows=${rows}`;
+  searchDatasets(
+    query: string,
+    filter: string = '',
+    start: number = 0,
+    rows: number = 12
+  ): Observable<{ results: PartialDataset[]; count: number }> {
+    const url = `${environment.backendUrl}/api/action/package_search?q=${encodeURIComponent(
+      query
+    )}&fq=${encodeURIComponent(filter)}&start=${start}&rows=${rows}`;
     return this.http.get<any>(url).pipe(
-      map(response => {
-
+      map((response) => {
         const items = response.result.results.map((item: any) => ({
           id: item.id,
           title: item.title,
           description: item.notes,
           modified: item.metadata_modified,
-          organizationName: item.organization.title,
-          publisher_name: item.publisher_name
+          organization: item.organization.title,
+          publisher_name: item.publisher_name,
+          tags: item.tags.map((tag: { name: string }) => tag.name),
+          theme: item.theme[0]
```

Comment:
I though this problem was fixed and this value was retrieve as an array, but if not - this is a string of array of strings - it whould be simpler running  `JSON.parse(intem.theme[0])`, no?